arduino/libraries/librariesresolver/cpp.go

```diff
@@ -212,7 +212,8 @@ func ComputePriority(lib *libraries.Library, header, arch string) int {
 	case libraries.User:
 		priority += 3
 	case libraries.Unmanaged:
-		priority += 4
+		// Bonus for libraries specified via --libraries flags, those libraries gets the highest priority
+		priority += 10000
```

Comment:
Did you consider this factor @cmaglie?:

https://github.com/arduino/arduino-cli/issues/2106#issue-1620849662

> Arduino Cloud does make significant use of "Unmanaged" locations:
>
> - The path to the globally installed libraries is specified via a `--libraries` flag
> - The path to each of the "CUSTOM" libraries are specified via a `--library` flag
> - The path to each of the sketch's pinned libraries are specified via a `--library` flag
>
> ...
>
> If the `--libraries` flag is to continue to be used to specified the paths to globally installed libraries in the Arduino Cloud compilation commands (vs. setting that path via `directories.user` as is standard), then a separate location type must be created for the `--library` and `--libraries` flags so that the weight increase can be bestowed only on `--library`.

Reply:
Thanks for the reminder!

I'd like to start a discussion with the Arduino Cloud team to see if the `directory.user` setting can be used (as it should be) without adding more complexity to the CLI.

Reply:
> Did you consider this factor @cmaglie?:
>
> https://github.com/arduino/arduino-cli/issues/2106#issue-1620849662
> 
> > Arduino Cloud does make significant use of "Unmanaged" locations:

We are now tracking this issue separately, so I'll proceed to merge this one in the meantime.